app/src/main/java/org/jellyfin/androidtv/details/TrailerPresenter.kt

```diff
@@ -0,0 +1,149 @@
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.base.IItemClickListener
+import org.jellyfin.androidtv.model.itemtypes.BaseItem
+import org.jellyfin.androidtv.model.itemtypes.PlayableItem
+import org.jellyfin.androidtv.ui.FavoriteBadge
+import org.jellyfin.androidtv.ui.MultiBadgeImageCardView
+import org.jellyfin.androidtv.ui.WatchedBadge
+import org.jellyfin.androidtv.util.ImageUtils
+import org.jellyfin.androidtv.util.dp
+import org.jellyfin.apiclient.model.entities.MediaUrl
+
+private const val SAFE_BRANDING_COMPLIANCE: Boolean = true
+
+class TrailerPresenter(private val context: Context, private val imageHeight: Int) : Presenter(), IItemClickListener {
+
+	val tldRegex = Regex("""^(.*\.|)(.+\...+)$""")
+	val youtubeDomains = arrayListOf("youtube.com", "youtu.be")
+
+	override suspend fun onItemClicked(item: Any?) {
+		requireNotNull(item)
+		if (item is BaseItem) {
+			val intent = Intent(context, DetailsActivity::class.java)
+			intent.putExtra("id", item.id)
+			context.startActivity(intent)
+		} else if (item is MediaUrl) {
+			val intent = Intent(Intent.ACTION_VIEW, Uri.parse(item.url))
+
+			if (intent.resolveActivity(context.packageManager) != null)
+				context.startActivity(intent)
+			else
+				Toast.makeText(context, R.string.toast_no_trailer_handler, Toast.LENGTH_LONG).show()
+		}
+	}
+
+	override fun onCreateViewHolder(parent: ViewGroup?): ViewHolder {
+		return ViewHolder(MultiBadgeImageCardView(ContextThemeWrapper(parent!!.context, R.style.MarqueeImageCardViewTheme)))
+	}
+
+	override fun onBindViewHolder(viewHolder: ViewHolder, item: Any) {
+
+		val cardView = (viewHolder.view as MultiBadgeImageCardView).apply {
+			isFocusable = true
+			isFocusableInTouchMode = true
+			setMainImageDimensions((ImageUtils.ASPECT_RATIO_16_9 * imageHeight).toInt(), imageHeight)
+			mainImageDrawable = TvApp.getApplication().getDrawableCompat(R.drawable.tile_port_video)
+		}
+
+		if (item is BaseItem) {
+			setupBaseItemCard(cardView, item)
+		} else if (item is MediaUrl) {
+			setupURLCard(cardView, item.name, Uri.parse(item.url))
+		} else {
+			throw IllegalArgumentException("Tried to present an unsupported item as trailer")
+		}
+	}
+
+	private fun isYoutubeUrl(uri: Uri) = youtubeDomains.contains(getDomain(uri))
+
+	private fun getDomain(uri: Uri): String? = uri.host?.let { host -> tldRegex.find(host)?.groups?.get(2)?.value }
+
+	private fun getProviderIcon(uri: Uri): Drawable? {
+		return if (isYoutubeUrl(uri) && !SAFE_BRANDING_COMPLIANCE)
+			context.getDrawable(R.drawable.ic_youtube)
+		else
+			null
+	}
+
+	private fun getThumbnailURL(uri: Uri): String? {
+		if (isYoutubeUrl(uri) && uri.queryParameterNames.contains("v")) {
+			return "https://img.youtube.com/vi/%s/0.jpg".format(uri.getQueryParameter("v"))
+		} else {
+			return null
+		}
+	}
+
+	private fun setupURLCard(cardView: MultiBadgeImageCardView, name: String, uri: Uri) {
```

Comment:
Add `cardView.setMainImageDimensions((4.0 / 3.0 * imageHeight).toInt(), imageHeight)`
Optionally add the aspect ratio to ImageUtils (4:3)

![image](https://user-images.githubusercontent.com/2305178/74928340-97eb2480-53d9-11ea-87ad-7a71aad846ed.png)

Reply:
We had a discussion in Matrix and another solution would be to still use a 16:9 based card but have the image "fit". This would remove the black bars for the 16:9 thumbnails but remove some pixels foor 4:3 thumbs